pkg/controllermanager/controller/seed/seed_backup_bucket_reconcile.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seed
+
+import (
+	"context"
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	gardencoreclientset "github.com/gardener/gardener/pkg/client/core/clientset/versioned"
+	gardencorelisters "github.com/gardener/gardener/pkg/client/core/listers/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/logger"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (c *Controller) backupBucketEnqueue(bb *gardencorev1beta1.BackupBucket) {
+	seedName := bb.Spec.SeedName
+	if seedName == nil {
+		return
+	}
+
+	c.seedBackupBucketQueue.Add(*seedName)
+}
+
+func (c *Controller) backupBucketAdd(obj interface{}) {
+	backupBucket, ok := obj.(*gardencorev1beta1.BackupBucket)
+	if !ok {
+		return
+	}
+
+	c.backupBucketEnqueue(backupBucket)
+}
+
+func (c *Controller) backupBucketUpdate(oldObj, newObj interface{}) {
+	oldBackupBucket, ok1 := oldObj.(*gardencorev1beta1.BackupBucket)
+	newBackupBucket, ok2 := newObj.(*gardencorev1beta1.BackupBucket)
+
+	if !ok1 || !ok2 {
+		return
+	}
+
+	if apiequality.Semantic.DeepEqual(oldBackupBucket.Spec, newBackupBucket.Spec) || apiequality.Semantic.DeepEqual(oldBackupBucket.Status, newBackupBucket.Status) {
```

Comment:
Hm, should this be 
```suggestion
	if !apiequality.Semantic.DeepEqual(oldBackupBucket.Spec, newBackupBucket.Spec) || !apiequality.Semantic.DeepEqual(oldBackupBucket.Status, newBackupBucket.Status) {
```

? Maybe I'm missing what this is supposed to do 👀